src/bin/cargo/commands/clean.rs

```diff
@@ -17,29 +21,243 @@ pub fn cli() -> Command {
         .arg(
             flag(
                 "dry-run",
-                "Display what would be deleted without deleting anything",
+                "Display what would be deleted without deleting anything (unstable)",
             )
             .short('n'),
         )
+
+        // NOTE: Not all of these options may get stabilized. Some of them are
+        // very low-level details, and may not be something typical users need.
+        .arg(
+            optional_opt(
+                "gc",
+                "Delete old and unused files (unstable) (comma separated): all, download, target, shared-target",
+            )
+            .hide(true)
+            .value_name("KINDS")
+            .value_parser(
+                PossibleValuesParser::new(["all", "download", "target", "shared-target"]).map(|x|
+                    match x.as_str() {
+                        "all" => AutoGcKind::All,
+                        "download" => AutoGcKind::Download,
+                        "target" => panic!("target is not yet implemented"),
+                        "shared-target" => panic!("shared-target is not yet implemented"),
+                        x => panic!("possible value out of sync with `{x}`"),
+                    }
+            ))
+            .require_equals(true),
```

Comment:
nit (since focus isn't on CLI yet): not normally a fan of requires_equals esp if we don't use it on everything

(looks like we only use it in one place so far, `--timings`)

Reply:
My thinking is that since it has an optional value, this helps ensure that there isn't a forwards compatibility hazard in case we decide if `cargo clean` should take standalone arguments. It currently doesn't, and I can't imagine it will, but I wanted to be cautious with it. 

I'm also uncertain how that would interact with subcommands (like maybe `cargo clean query`). I'm guessing the ambiguity there is also not important?

Mainly I wanted to be careful since optional arguments can be ambiguous.

I'm not sure if there is a better design here for the CLI option. We could change it to require an argument, or switch to separate CLI options, or maybe remove the options to simplify it. 🤷

Reply:
Since the CLI is not being decided on at this time, we can pass over this.